File: Company-gRPC-service/src/main/java/grpc/project/companygrpcservice/Department/Department.java
package grpc.project.companygrpcservice.Department;

import grpc.project.companygrpcservice.Employee.Employee;
import grpc.project.companygrpcservice.Project.Project;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.util.Collection;
@Data @AllArgsConstructor @NoArgsConstructor @Builder
public class Department {
    @Id
    private String D_no;
    private String name;
    private String speciality;
    private Collection<Employee> employees;
    private Employee manager;
    private Collection<Department> subDepartments;
    private Collection<Project> projects;
}

File: Employee-gRPC-service/Models/Departments/Department.cs
namespace Employee_gRPC_service.Models {
    public class Department {
    public string? D_no{get; set;}
    private string? name{get; set;}
    private string? speciality{get; set;}
    private ICollection<Employee>? employees{get; set;}
    private Employee? manager{get; set;}
    private ICollection<Department>? subDepartments{get; set;}
    private ICollection<Project>? projects{get; set;}
    }
}